code/data_structures/soft_heap/soft_heap.h

```diff
@@ -0,0 +1,259 @@
+/**
```

Comment:
Kindly correct the extension of this code 👍

Reply:
Sure, but a library of data structures as C++ source files are much more difficult to use in other projects than just a header.

Reply:
@JonNRb AdiChat is correct, since you have defined a main function here, meaning this cannot be used as a library anyway.

Reply:
I am well aware, which is why there was a `#if 1` that you could easily make `#if 0`. All of the data structures that people are writing for Cosmos *absolutely* need tests, but we don’t have any packaging or build system to define how to link things together.

(github email replys don't seem to attach to the right thread...)

Since most code is C++, Java, and Python, we could be using something like https://bazel.build/ to add unit tests and let people leverage the code in this repository without just copying the raw files. I'd be happy to investigate something like this for the existing codebase.